ReactiveCocoa/Swift/Property.swift

```diff
@@ -5,69 +5,182 @@ import enum Result.NoError
 public protocol PropertyType {
 	associatedtype Value
 
-	/// The current value of the property.
-	var value: Value { get }
-
 	/// A producer for Signals that will send the property's current value,
-	/// followed by all changes over time.
+	/// followed by all changes over time, then complete when the property has
+	/// deinitialized.
 	var producer: SignalProducer<Value, NoError> { get }
 
 	/// A signal that will send the property's changes over time.
 	var signal: Signal<Value, NoError> { get }
+
+	/// Performs an arbitrary action using the current value of the
+	/// variable. Conforming types may optionally guarantee atomicity.
+	///
+	/// Returns the result of the action.
+	func withValue<Result>(@noescape action: Value throws -> Result) rethrows -> Result
```

Comment:
`MutableProperty` has `withValue` (and `modify`) so that the supplied action block is executed within the critical section. But the `AnyProperty` wrapper did not expose it.